File: java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        int pow = String.valueOf(numberToCheck).length();
        int sum = 0;
        int temp = numberToCheck;
        while (temp > 0){
            int reminder = temp % 10;
            // raise by pow
            sum += Math.pow(reminder, pow);
            // floor by 10
            temp = Math.floorDiv(temp, 10);
        }
        if (numberToCheck == sum){
            return true;
        } else{
            return false;
        }

    }

}

File: go/armstrong-numbers/armstrong_numbers.go
package armstrong

import (
	"math"
	"strconv"
)

func IsNumber(n int) bool {
	var pow = len([]rune(strconv.Itoa(n)));
	sum := 0;
	temp := n;

	for temp > 0 {
		var reminder = temp % 10
		// raised by pow
		sum += int(math.Pow(float64(reminder), float64(pow)));
		// floor by 10
		temp = temp / 10;
	}

	if (n == sum){
		return true;
	} else {
		return false;
	}
}